src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -193,16 +206,43 @@ public Action<ClusterBuilder> ClusterConfigurator
             }
         }
 
+#pragma warning disable CS0618
+        internal ClusterConnectionModeSwitch ClusterConnectionModeSwitch
+#pragma warning restore CS0618
+        {
+            get
+            {
+                return _clusterConnectionModeSwitch;
+            }
+        }
+
         /// <summary>
         /// Gets or sets the connection mode.
         /// </summary>
+        [Obsolete("Use DirectConnection instead.")]
         public ConnectionMode ConnectionMode
         {
-            get { return _connectionMode; }
+            get
+            {
+                if (_clusterConnectionModeSwitch == ClusterConnectionModeSwitch.UseDirectConnection)
+                {
+                    throw new InvalidOperationException("MongoClientSettings.ConnectionMode cannot be used when ClusterConnectionModeSwitch.UseDirectConnection.");
```

Comment:
Remove message to:

    "ConnectionMode cannot be used when ConnectionModeSwitch is set to UseDirectConnection."

Find all other places and reword to match (including analogous messages for DirectConnection).

Reply:
done